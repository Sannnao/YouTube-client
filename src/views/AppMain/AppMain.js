import Data from './MainData';

export default class AppMain {
  constructor(dataArr) {
    this.dataArr = dataArr;
  }

  static render() {
    const main = document.createElement('main');
    const contentContainer = document.createElement('div');

    main.setAttribute('class', 'main');
    contentContainer.setAttribute('class', 'content-container');

    main.appendChild(contentContainer);
    document.body.appendChild(main);
  }

  fill() {
    const contentContainer = document.querySelector('.content-container');

    const clipContainer = new Data(this.dataArr);
    const clipContainerArr = clipContainer.getContainerArr();

    clipContainerArr.map(elem => contentContainer.appendChild(elem));
  }
}
