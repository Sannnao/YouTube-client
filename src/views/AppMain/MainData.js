export default class Data {
  constructor(dataArr) {
    this.dataArr = dataArr;
  }

  getContainerArr() {
    return this.dataArr.map((elem) => {
      const container = document.createElement('section');
      const title = document.createElement('h2');
      const anchor = document.createElement('a');
      const imgAnchor = document.createElement('a');
      const picture = document.createElement('img');
      const channelName = document.createElement('p');
      const date = document.createElement('p');
      const viewCount = document.createElement('p');
      const description = document.createElement('p');

      container.setAttribute('class', 'clip-container');

      title.setAttribute('class', 'title-item');

      anchor.setAttribute('href', `https://www.youtube.com/watch?v=${elem.clipId}`);
      anchor.setAttribute('title', 'to YouTube');
      anchor.setAttribute('target', '_blank');

      imgAnchor.setAttribute('href', `https://www.youtube.com/watch?v=${elem.clipId}`);
      imgAnchor.setAttribute('title', 'to YouTube');
      imgAnchor.setAttribute('target', '_blank');
      imgAnchor.setAttribute('class', 'img-anchor');

      picture.setAttribute('class', 'clip-picture');
      picture.setAttribute('src', `${elem.picture.medium.url}`);
      picture.setAttribute('width', `${elem.picture.medium.width}`);
      picture.setAttribute('height', `${elem.picture.medium.height}`);
      picture.setAttribute('alt', 'clip-picture');

      channelName.setAttribute('class', 'channel-item clip-item-container');
      date.setAttribute('class', 'date-item clip-item-container');
      viewCount.setAttribute('class', 'view-count-item clip-item-container');
      description.setAttribute('class', 'description-item');

      anchor.innerHTML = elem.title;
      channelName.innerHTML = elem.channelName;
      [date.innerHTML] = elem.date.split('T');
      viewCount.innerHTML = elem.viewCount;
      description.innerHTML = elem.description;

      imgAnchor.appendChild(picture);
      title.appendChild(anchor);
      container.appendChild(title);
      container.appendChild(imgAnchor);
      container.appendChild(channelName);
      container.appendChild(date);
      container.appendChild(viewCount);
      container.appendChild(description);

      return container;
    });
  }
}
