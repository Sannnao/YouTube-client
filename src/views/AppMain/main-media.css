@media (max-height: 768px) {
  .content-container {
    height: 100%;
  }
}

@media (max-width: 1400px) {
  .clip-container {
    width: 27.3333333%;
    margin: 0 3%;
  }
}

@media (max-width: 1300px) {
  .clip-container {
    width: 29.3333333%;
    margin: 0 2%;
  }
}

@media (max-width: 1100px) {
  .clip-container {
    width: 42%;
    margin: 0 4%;
  }
}

@media (max-width: 900px) {
  .clip-container {
    width: 46%;
    margin: 0 2%;
  }
}

@media (max-width: 700px) {
  .clip-container {
    width: 70%;
    margin: 0 15%;
  }
}

@media (max-width: 600px) {
  .clip-container {
    width: 76%;
    margin: 0 12%;
  }
}

@media (max-width: 500px) {
  .clip-container {
    width: 88%;
    margin: 0 6%;
  }
}

@media (max-width: 400px) {
  .clip-container {
    width: 96%;
    margin: 0 2%;
    font-size: 14px;
  }

  .title-item {
    height: 80px;
    padding: 10px;
    font-size: inherit;
  }

  .clip-picture {
    height: 170px;
    margin-bottom: 10px;
  }
}
