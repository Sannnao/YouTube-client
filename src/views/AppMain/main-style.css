.main {
  display: flex;
  align-items: center;
  overflow-x: hidden;
}

.content-container {
  position: relative;
  display: flex;
  width: 100%;
  height: 82%;
  transition: transform 1s;
}

.main:hover {
  cursor: grab;
}

.main.active {
  cursor: grabbing;
  cursor: -webkit-grabbing;
}

.clip-container {
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  align-items: center;
  flex-shrink: 0;
  width: 23%;
  height: 100%;
  margin: 0 1%;
}

.title-item {
  display: flex;
  justify-content: center;
  align-items: center;
  width: 100%;
  height: 90px;
  padding: 10px 20px;
  text-align: center;
  font-size: 18px;
  color: #F2F4F3;
  background-color: #49111C;

  box-shadow: -1px -1px 5px 0 rgb(190, 190, 190);
}

.title-item:hover {
  color: rgb(253, 245, 133);
}

.img-anchor {
  display: inline-flex;
  width: 100%;
  transition: transform 1s;
  transform: scale(1, 1);

  box-shadow: -1px -1px 5px 0 rgb(190, 190, 190);
}

.img-anchor:hover {
  transition: transform 1s;
  transform: scale(1.05, 1.05);
}

.clip-picture {
  width: 100%;
  height: 180px;
}

.clip-item-container {
  display: flex;
  justify-content: center;
  align-items: center;
  width: 100%;
  height: 40px;
  padding: 0 5%;
  background-position: 4% 50%;
  background-repeat: no-repeat;
  background-size: 20px;
  box-shadow: -1px -1px 5px 0 rgb(190, 190, 190);
}

.channel-item {
  padding: 0 20px;
  overflow: auto;
  background-image: url('../../assets/icons/channel-icon.png');
}

.date-item {
  background-image: url('../../assets/icons/date.png');
}

.view-count-item {
  background-image: url('../../assets/icons/view.png');
}

.description-item {
  width: 100%;
  height: 200px;
  padding: 0 5%;
  overflow: auto;
  font-size: 14px;
  font-style: italic;
  word-wrap: break-word;
  word-break: break-all;

  box-shadow: -1px -1px 5px 0 rgb(190, 190, 190);
}
