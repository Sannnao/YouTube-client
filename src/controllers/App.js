import AppModel from '../models/AppModel';
import AppMain from '../views/AppMain';

export default class App {
  constructor(request, key) {
    this.searchUrl = `https://www.googleapis.com/youtube/v3/search?fields=nextPageToken,items/id/videoId&part=snippet&key=${key}&type=video&maxResults=15&q=${request}`;
    this.videoUrl = null;
    this.request = request;
    this.key = key;
  }

  async start() {
    const searchModel = new AppModel(this.searchUrl);
    const searchData = await searchModel.getSearchData();
    const { pageToken, videoId } = searchData;


    this.searchUrl = `https://www.googleapis.com/youtube/v3/search?fields=nextPageToken,items/id/videoId&part=snippet&key=${this.key}&type=video&maxResults=15&q=${this.request}&pageToken=${pageToken}`;

    this.videoUrl = `https://www.googleapis.com/youtube/v3/videos?fields=items(id,snippet(publishedAt,title,description,thumbnails/medium,channelTitle),statistics/viewCount)&part=snippet,statistics&key=${this.key}&id=${videoId.join(',')}`;

    const videoModel = new AppModel(this.videoUrl);
    const videoData = await videoModel.getVideoData();

    const main = new AppMain(videoData);

    main.fill();
  }
}
