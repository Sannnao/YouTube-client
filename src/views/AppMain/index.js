import AppMain from './AppMain';
import './main-style.css';
import './main-media.css';

export default AppMain;
