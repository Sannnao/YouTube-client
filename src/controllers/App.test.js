import App from './App';
import AppMain from '../views/AppMain/AppMain';

describe('App.contructor', () => {
  describe('App.searchUrl', () => {
    it('Should be search URL from YouTube Data API', () => {
      const app = new App('request', 'key');

      expect(app.searchUrl.split('?')[0]).toEqual('https://www.googleapis.com/youtube/v3/search');
    });
  });

  describe('App.videoUrl', () => {
    it('Should be video URL from YouTube Data API', () => {
      const app = new App();
      app.videoUrl = 'https://www.googleapis.com/youtube/v3/videos?fields=items(id,snippet(publishedAt,title,description,thumbnails/medium,channelTitle),statistics/viewCount)&part=snippet,statistics&key=&id=';

      expect(app.videoUrl.split('?')[0]).toEqual('https://www.googleapis.com/youtube/v3/videos');
    });
  });
});

describe('App.start', () => {
  it('Should render content correctly', () => {
    const app = new App('req', 'key');
    AppMain.render();
    app.start();

    expect(document.querySelector('.content-container').innerHTML).toMatchSnapshot();
  });
});
