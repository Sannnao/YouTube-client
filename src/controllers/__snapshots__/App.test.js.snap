// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.start Should render content correctly 1`] = `""`;
